docs/emulate-and-dev-in-docker.md

```diff
@@ -0,0 +1,169 @@
+- ⚙️ **Makefile integration**: Use `make emulate` to build and sync in one step
+
+#### Option 1: Manual Sync via `sync_to_emulator`
+We provide a helper command called `sync_to_emulator`, which simplifies the
+process of syncing the build outputs to the emulation environment.
+Run the following commands inside the container:
+```bash
+sync_to_emulator -ta $TA
+sync_to_emulator -host $HOST_APP
+```
+Run sync_to_emulator -h for more usage options.
+
+#### Option 2: Use the Makefile's `emulate` Target
+For convenience during daily development, this syncing step is integrated into
+the Makefile under the `emulate` target. This will automatically build the
+artifacts and sync them to the emulator in one step:
+```bash
+make -C examples/hello_world-rs/ emulate
+```
+We recommend using this approach for a streamlined development workflow.
+
+## 4. Multi-Terminal Execution
+
+The emulation workflow requires three additional terminals to monitor
+various aspects of the system:
+
+- **Terminal B**: 🖥️ **Normal World Listener** - Provides access to the guest VM shell
+- **Terminal C**: 🔒 **Secure World Listener** - Monitors Trusted Application output logs  
+- **Terminal D**: 🚀 **QEMU Control** - Controls the QEMU emulator
+
+Built-in commands are provided in the Docker image. These commands are located
+in `/opt/teaclave/bin/` and are included in the default user's $PATH.
+
+You may use `bash -l` or the full path when executing with docker exec.
+
+**Terminal B** (Guest VM Shell):
+```bash
+# Connect to the guest VM shell for running commands inside the emulated environment
+$ docker exec -it teaclave_dev_env bash -l -c listen_on_guest_vm_shell
+
+# Alternative: Use full path
+$ docker exec -it teaclave_dev_env /opt/teaclave/bin/listen_on_guest_vm_shell
+```
+
+**Terminal C** (Secure World Output Monitor):
+```bash
+# Monitor Trusted Application output logs in real-time
+$ docker exec -it teaclave_dev_env bash -l -c listen_on_secure_world_log
+
+# Alternative: Use full path  
+$ docker exec -it teaclave_dev_env /opt/teaclave/bin/listen_on_secure_world_log
+```
+
+### 5. Start the Emulation
+
+After the listeners are set up, we can start the QEMU emulator.
+
+**Terminal D** (QEMU Control):
+```bash
+# Launch QEMU emulator with debug output and connect to monitoring ports
+$ docker exec -it teaclave_dev_env bash -l -c "LISTEN_MODE=ON start_qemuv8"
+```
+
+> ⏳ **Wait for the QEMU environment to fully boot...** 
+You should see boot messages in Terminal D and the guest VM shell prompt 
+in Terminal B.
+
+After QEMU in Terminal D successfully launches, switch to Terminal B, which
+provides shell access to the guest VM's normal world.
+
+**Terminal B** (Inside Guest VM):
+From this shell, you’ll notice that the artifacts synced in **Step 3** are
+already available in the current working directory. This shared directory is
+mounted between the host and the guest VM, so any changes made—either inside
+or outside QEMU—are immediately reflected on both sides.
```

Comment:
Add: The `ta/` directory is mapped to `/lib/optee_armtz`, which is the default directory OP-TEE uses to search for TAs.